src/Layout/Header/DeviceSelector/DeviceSelector.js

```diff
@@ -1,7 +1,99 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+/** Actions */
+import { fetchDevicesAction, changeDeviceAction } from '@context/spotify';
 /** Components */
-import Select from '@common/Select';
+import { Popover, PopoverTrigger } from '@common/Popover';
+import { ReactComponent as DevicesIcon } from '@assets/svg/devices.svg';
+import DeviceSelectorItem from './DeviceSelectorItem';
+/** Styled components */
+import {
+  DevicesSelectorWrapper,
+  DevicesSelectorItemsWrapper,
+  NoDevicesFoundText,
+} from './DeviceSelector.styled';
 
-const Avatar = () => <Select />;
+const DeviceSelector = ({ intl }) => {
+  const [forceClose, setForceClose] = useState(false);
 
-export default Avatar;
+  const {
+    spotify: { devices, activeDevice },
+    playing: { current },
+  } = useSelector(state => state);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    activeDevice && dispatch(fetchDevicesAction());
+  }, [activeDevice]);
+
+  const changeDeviceHandler = deviceId => {
+    dispatch(changeDeviceAction(deviceId, current));
+    setForceClose(true);
+  };
+
+  const openSelectorHandler = () => {
+    dispatch(fetchDevicesAction());
+  };
+
+  const closeSelectorHandler = () => {
+    setForceClose(false);
+  };
+
+  const hasActiveDevice = () => {
+    return !!devices.find(device => device.is_active);
```

Comment:
`return Array.isArray(devices) && !!devices.find(device => device.is_active);` 

Although in the store it initializes as an empty array, so its up to you, its only a single comment.

Reply:
@MateoBeMo As you said, I initialize `devices` as an empty array in the store, so I don't really see the point of using Array.isArray.

In any case, I have a question. **hasActiveDevice** is a function that returns a boolean and I use it in the **isActive** prop from `DevicesSelectorWrapper`. How clean and efficient is that? Is there a better solution?

Reply:
For decouple the deviceSelector with the store bro, in the future the global state  implementation could change, and the initialize could be different. But, its ok, if that happens, I'll find you and I' ll kill you.

Good question, we should investigated, because it can be a boolean variable too.
`const hasActiveDevice = !!devices.find(device => device.is_active);`

Reply:
I think (maybe im wrong) that, when we use the function way, each render will init new  **hasActiveDevice** function, and will pass it to the child as prop. It means, that the child will be re-rendered because the reference has changed.

Reply:
> For decouple the deviceSelector with the store bro, in the future the global state implementation could change, and the initialize could be different. But, its ok, if that happens, I'll find you and I' ll kill you.

That's why you have to be careful when changing the initialization of a global variable that is already in use 😉

> I think (maybe im wrong) that, when we use the function way, each render will init new **hasActiveDevice** function, and will pass it to the child as prop. It means, that the child will be re-rendered because the reference has changed.

Yes, I think it might be a re-rendering issue there. Maybe it's time to use React Memoization to avoid re-rendering leaks!